File: src/lib/wmo.ts
// Map Open-Meteo WMO weather codes to our internal icon kinds.
// https://open-meteo.com/en/docs

export type WeatherKind =
  | "sun"
  | "partly"
  | "cloud"
  | "fog"
  | "rain"
  | "snow"
  | "thunder";

export function wmoToKind(code: number): WeatherKind {
  if (code === 0) return "sun";
  if (code === 1 || code === 2) return "partly";
  if (code === 3) return "cloud";
  if (code === 45 || code === 48) return "fog";
  if (code >= 51 && code <= 67) return "rain";
  if (code >= 80 && code <= 82) return "rain";
  if (code >= 71 && code <= 77) return "snow";
  if (code === 85 || code === 86) return "snow";
  if (code >= 95) return "thunder";
  return "cloud";
}

export function wmoLabel(code: number): string {
  const m: Record<number, string> = {
    0: "clear sky",
    1: "mostly clear",
    2: "partly cloudy",
    3: "overcast",
    45: "fog",
    48: "rime fog",
    51: "light drizzle",
    53: "drizzle",
    55: "dense drizzle",
    56: "freezing drizzle",
    57: "freezing drizzle",
    61: "light rain",
    63: "rain",
    65: "heavy rain",
    66: "freezing rain",
    67: "freezing rain",
    71: "light snow",
    73: "snow",
    75: "heavy snow",
    77: "snow grains",
    80: "rain showers",
    81: "rain showers",
    82: "violent showers",
    85: "snow showers",
    86: "snow showers",
    95: "thunderstorm",
    96: "thunderstorm w/ hail",
    99: "thunderstorm w/ hail",
  };
  return m[code] ?? "unknown";
}

File: android/app/src/main/java/dev/pi0trdotsys/homebrewweather/widget/Wmo.kt
package dev.pi0trdotsys.homebrewweather.widget

/**
 * Direct Kotlin port of src/lib/wmo.ts — maps Open-Meteo WMO weather codes to
 * the same icon "kind" strings used by [PixelIcons].
 */
object Wmo {
    fun wmoToKind(code: Int): String {
        if (code == 0) return "sun"
        if (code == 1 || code == 2) return "partly"
        if (code == 3) return "cloud"
        if (code == 45 || code == 48) return "fog"
        if (code in 51..67) return "rain"
        if (code in 80..82) return "rain"
        if (code in 71..77) return "snow"
        if (code == 85 || code == 86) return "snow"
        if (code >= 95) return "thunder"
        return "cloud"
    }
}
